### PR TITLE
[Validation] DisconnectBlock updates.

### PR DESCRIPTION
**Pull request by furszy**
Built on top of #1733.

Straightforward PR, no functional changes. Purely code cleanup and refactoring around the `DisconnectBlock` function.

### CODE REVIEW

**Comment by random-zebra** [comment]
Can be rebased now (#1733 merged).

**Comment by furszy** [comment]
yep, doing it

**Comment by furszy** [comment]
rebased

File: src/main.cpp [hunk 1]
```diff
@@ -1949,41 +1949,55 @@ static bool ApplyTxInUndo(const CTxInUndo& undo, CCoinsViewCache& view, const CO
...
+    }
+    if (!blockUndo.ReadFromDisk(pos, pindex->pprev->GetBlockHash())) {
+        error("DisconnectBlock() : failure reading undo data");
+        return DISCONNECT_FAILED;
+    }
 
-    if (blockUndo.vtxundo.size() + 1 != block.vtx.size())
-        return error("DisconnectBlock() : block and undo data inconsistent");
+    if (blockUndo.vtxundo.size() + 1 != block.vtx.size()) {
+        error("DisconnectBlock() : block and undo data inconsistent");
```
Comment (random-zebra) [inline src/main.cpp:1985]: nit: use `__func__` instead of `DisconnectBlock()` in the error messages.

Comment (furszy) [reply src/main.cpp]: done

File: src/main.cpp [hunk 2]
```diff
@@ -1916,52 +1916,96 @@ static bool AbortNode(CValidationState& state, const std::string& strMessage, co
...
+    }
+    if (!blockUndo.ReadFromDisk(pos, pindex->pprev->GetBlockHash())) {
+        error("%s: failure reading undo data", __func__);
+        return DISCONNECT_FAILED;
+    }
 
-    if (blockUndo.vtxundo.size() + 1 != block.vtx.size())
-        return error("DisconnectBlock() : block and undo data inconsistent");
+    if (blockUndo.vtxundo.size() + 1 != block.vtx.size()) {
+        error("%s: block and undo data inconsistent", , __func__);
```
Comment (random-zebra) [inline src/main.cpp:1985]: extra `,`

Comment (furszy) [reply src/main.cpp]: oops, doing two things at the same time. Pushing again.

**Comment by furszy** [comment]
rebased, had conflicts with recently merged #1746

**Pull request merged**